q/compare.go

```diff
@@ -48,6 +48,18 @@ func compare(a, b interface{}, tok token.Token) bool {
 
 			return constant.Compare(constant.MakeFloat64(vala.Float()), tok, constant.MakeFloat64(bla))
 		}
+	case ak == reflect.Struct:
```

Comment:
I think you can start with two simple tests:
- Create a custom struct A and compare 2 instances of A with `q.Eq`. 2 instances with exactly the same values must match
- Create another custom struct B with exactly the same fields as A and compare an instance of A with an instance of B with `q.Eq`. 2 instances with the same values for the same fields must match.

Of course, you can name the structs as you like :)